src/Components/Form/Form.js

```diff
@@ -1,39 +1,68 @@
-import React, { useState, useContext } from 'react';
+import React, { useState, useContext, useRef} from 'react';
 import axios from 'axios';
 import { SearchButton } from './SearchButton';
 import { HotelsSearch } from './HotelsSearch';
 import { apiUrl } from '../../Services/Constanst/links';
 import { AvailableHotelsContext } from '../../Context/AvailableHotelsContext';
 import { Calendar } from '../Calendar';
+import { UsersFilter } from './UsersFilter';
+import { useClickOutside } from '../../Hooks/useClickOutSide';
 
 import './Form.css';
+import { fetchData, getData, wrapPromise } from '../../lib/wrapPromise';
+import { getHotels } from '../../Services/HotelsCards/SearchAPI';
 
 export const Form = () => {
   const [formState, setFormState] = useState('');
+  const [filterActive, setFilterActive] = useState(false);
   const { setAvailable } = useContext(AvailableHotelsContext);
+  const AdRef = useRef();
 
   function handleSubmit(e) {
     e.preventDefault();
 
-    formState !== '' &&
-      axios
-        .get(`${apiUrl}`, {
-          params: {
-            search: `${formState}`,
-          },
-        })
-        //TODO: Error handling, loader
-        .then(formState === '' ? false : (resp) => setAvailable(resp.data))
-        .catch(function (error) {
-          console.log(error);
-        });
+    apiUrl.searchParams.set('search', `${formState}`);
+
+    const hotels = wrapPromise(fetchData(apiUrl));  
```

Comment:
look, for the first submit your component catch status of promise in pending, so you need to set Data only after promise will be finished. 
 your function must be async in this case
````
 async function handleSubmit(e) {
    e.preventDefault();
    apiUrl.searchParams.set('search', `${formState}`);
     const hotels =  wrapPromise(getData(apiUrl));
   try {
     const availableHotels = await hotels;
    formState !== '' && setAvailable(availableHotels);

     //   formState !== '' &&
     //  axios
     //   .get(`${apiUrl}`, {
     //     params: {
     //       search: `${formState}`,
     //     },
     //   })
     //   // TODO: Error handling, loader
     //   .then((resp) => setAvailable(resp.data))
     //   .catch(function (error) {
     //     console.log(error);
     //   });
   }
   catch (error){

   }
    setFormState('');
  }
````